cassandra/src/main/java/io/micronaut/configuration/cassandra/CassandraConfiguration.java

```diff
@@ -26,21 +25,25 @@
  * used to denote the primary cluster bean.
  *
  * @author Nirav Assar
+ * @author Michael Pollind
  * @since 1.0
  */
-@EachProperty(value = "cassandra", primary = "default")
+@EachProperty(value = CassandraConfiguration.PREFIX, primary = "default")
 public class CassandraConfiguration {
+    public static final String PREFIX = "cassandra.datasource";
+    private final String name;
 
-    @SuppressWarnings("WeakerAccess")
-    @ConfigurationBuilder(allowZeroArgs = true, prefixes = { "with", "add" })
```

Comment:
Perhaps I'm missing something; what's replacing this annotation as a means of building the connection details from the micronaut config (like `application.yml`)

Reply:
so they don't have an object that you can just map the configs to easily. Before this I had a large amount of static classes that I was trying to map configuration to but that posed too much of a challenge.  I manually resolve that in the CassandraSessionFactory where I use the cassandra.datasource.<name>.<config_path> where config_path is a path define in DefaultDriverOption. The values I chose for the default are from this reference: https://github.com/datastax/java-driver/blob/4.0.1/core/src/main/resources/reference.conf

the datastax driver 4.4.0 is using this to define configurations: https://github.com/lightbend/config

Reply:
Are there comparable properties to the ones that would have been provided previously (like `cassandra.default.port` and the like provided in the tests?)

Reply:
actually, no I don't think so. Any suggestions how I would go about this, I can move it back to the old path and add some properties so the cassandra driver would match. I'm not even sure if this will drop into place with the old driver. It looks like the whole thing was rewritten from 3 to 4. I can check those old paths and remap those properties, but wouldn't cassandra.datasource map back to the old config with cassandra.default.